packages/bundler-plugin-core/src/types.ts

```diff
@@ -241,9 +241,25 @@ export interface Options {
      * Defaults to `false`.
      */
     injectBuildInformation?: boolean;
+
+    /**
+     * Metadata associated with this module.
+     * The metadata is serialized and can be looked up at runtime by URL.
+     *
+     * Note: This option is currently only supported by `@sentry/webpack-plugin`.
+     */
+    moduleMetadata?: object | ModuleMetadataCallback;
```

Comment:
Hm. I would like to make this `any | ModuleMetadataCallback` but it's just gonna get turned into `any`.

I think we should just do it and people who follow the definition will see that this also accepts a callback. We should add to the jsdoc that this can be a static value or a callback.

Reply:
```ts
type MetadataValue = object | boolean | number | string | symbol`;

moduleMetadata?: MetadataValue | ModuleMetadataCallback;
```
😆